src/vulkan/engine_vulkan.cc

```diff
@@ -192,6 +192,18 @@ Result EngineVulkan::CreatePipeline(amber::Pipeline* pipeline) {
 
   info.vk_pipeline = std::move(vk_pipeline);
 
+  // Set the entry point names for the pipeline.
+  for (const auto& shader_info : pipeline->GetShaders()) {
+    VkShaderStageFlagBits stage = VK_SHADER_STAGE_FLAG_BITS_MAX_ENUM;
+    r = ToVkShaderStage(shader_info.GetShaderType(), &stage);
+    if (!r.IsSuccess())
+      return r;
+    const auto& name = shader_info.GetEntryPoint();
+    if (!name.empty()) {
```

Comment:
If it is empty, it would be better to just return an error here?

```
if (name.empty())
  return Result("Vulkan: pipeline entry point is empty");
info.vk_pipeline->SetEntryPointName(stage, name);
```

Reply:
It is valid to not set an entry name if you plan to use the default name.